.eslintrc.js

```diff
@@ -54,37 +54,50 @@ module.exports = {
 		'lines-between-class-members': [1],
 		'max-len': [0],
 		'newline-per-chained-call': [0, { 'ignoreChainWithDepth': 2 }],
-		'no-mixed-operators': [0],
+		'no-mixed-operators': [1],
 		'no-nested-ternary': [0],
-		'no-param-reassign': [0], // TODO: enable this check
+		'no-param-reassign': ['error', {
+			props: true,
+			ignorePropertyModificationsFor: [
+				'acc', // for reduce accumulators
+				'document', // for IIFE with window & document params
+				'c2pFrame', // for mutable iframe c2p element
+				'trackerEl', // for trackers.forEach()
+				'categoryEl', // for categories.forEach()
+				'dataEntry', // for data.forEach()
+				'appEntry', // for apps.forEach()
+				'sEntry', // for apps.forEach()
+				'confMutable', // for mutable conf settings
+				'eventMutable', // for mutable event details objects
+			]
+		}],
 		'no-plusplus': [0],
-		'no-prototype-builtins': [0], // TODO: enable this check
-		'no-restricted-syntax': [0], // TODO: enable this check
+		'no-prototype-builtins': [0],
+		'no-restricted-syntax': [1],
```

Comment:
Remove 'no-restricted-syntax'

Reply:
updated in feature/cleanup-cleanup as this PR was already merged.

Reply:
lgtm in feature/cleanup-cleanup